File: udemy/ds-and-a/csharp-solutions/DSAndA.Test/LinkedLists/Assignment6/IsCircular.Test.cs
using DSAndA.LinkedLists;
using Xunit;

namespace DSAndA.Test.LinkedLists.Assignment6
{
    public class IsCircularTestFixture
    {
        [Fact]
        public void IsCircularTest1()
        {
            LinkedList<int> linkedList = new LinkedList<int>();
            linkedList.InsertAtHead(3);
            linkedList.InsertAtHead(2);
            linkedList.InsertAtHead(1);

            Node<int> head = linkedList.Search(1);
            Node<int> last = linkedList.Search(3);
            last.NextNode = head;

            Assert.True(linkedList.IsCircular());
        }

        [Fact]
        public void IsCircularTest2()
        {
            LinkedList<int> linkedList = new LinkedList<int>();
            linkedList.InsertAtHead(3);
            linkedList.InsertAtHead(2);
            linkedList.InsertAtHead(1);

            Assert.False(linkedList.IsCircular());
        }
    }
}

File: udemy/ds-and-a/javascript-solutions/src/tests/LinkedLists/Assignment6/IsCircular.test.js
const LinkedList = require("../../../LinkedLists/LinkedList");

test('given a circular list, should return true', () => {
    let linkedList = new LinkedList();
    let tail = linkedList.insertAtHead(2);
    linkedList.insertAtHead(1);
    linkedList.insertAtHead(0);

    tail.nextNode = linkedList.head;

    expect(linkedList.isCircular()).toBeTruthy();
});

test('given a list which is not circular, should return false', () => {
    let linkedList = new LinkedList();
    linkedList.insertAtHead(2);
    linkedList.insertAtHead(1);
    linkedList.insertAtHead(0);

    expect(linkedList.isCircular()).toBeFalsy();
});